apps/admin/frontend/src/screens/write_ins_adjudication_screen.tsx

```diff
@@ -306,6 +316,17 @@ export function WriteInsAdjudicationScreen(): JSX.Element {
   const nextButton = useRef<Button>(null);
   const firstAdjudicationControl = useRef<HTMLFieldSetElement>(null);
 
+  const writeInAdjudicationContext = writeInAdjudicationContextQuery.data;
+  const currentWriteIn = writeInAdjudicationContext?.writeIn;
+  const currentWriteInMarkedInvalid =
+    currentWriteIn &&
+    currentWriteIn.status === 'adjudicated' &&
+    currentWriteIn.adjudicationType === 'invalid';
+
+  const writeInImageView = writeInImageViewQuery.data;
+  const hmpbWriteInImageView = writeInImageView as HmpbWriteInImageView;
+  const bmdWriteInImageView = writeInImageView as BmdWriteInImageView;
```

Comment:
I found it a little confusing to cast `writeInImageView` to both `Hmpb` and `Bmd` types and then check their fields as a way of determining which one it is.

Wondering if we could do something like:

```ts
const isHmpb = 'ballotCoordinates' in writeInImageView;
```

and then use `isHmpb` lower down when deciding what to render

Reply:
Yeah I wasn't completely sure what the best way to do this was, will play with that idea.